### ISSUE TITLE
Redesign state management

### ISSUE THREAD

**Issue by solid-yuriiprykhodko** [DenisManuch/todo_app#17]
The app is relatively small, but definitely needs some state management solution.

You most certainly can keep things vanilla and use only Flutter library, but it already calls for some `InheritedWidget` use, aside from your usual `setState` calls. This is a good exercise, but it's not necessarily worth the time.

I'd suggest using Provider + ChangeNotifier. It will make fixing things like #11 a breeze.

**Comment by DenisManuch** [comment]
Yes, I was thinking about this yesterday. The logic needs to be changed.
